lib/phoenix/transports/long_poll.ex

```diff
@@ -99,8 +99,8 @@ defmodule Phoenix.Transports.LongPoll do
       "phx:lp:"
       <> Base.encode64(:crypto.strong_rand_bytes(16))
       <> (System.system_time(:millisecond) |> Integer.to_string)
-
-    arg = {endpoint, handler, opts, conn.params, priv_topic}
+    meta = Transport.extract(conn, Keyword.get(opts, :extract, []))
```

Comment:
What do you think about using `:meta` instead of `:extract` here?

```elixir
socket "/ws", UserSocket,
      websocket: [
        check_origin: ["//example.com"],
        timeout: 200,
        meta: [:x_headers, :peer_data]
      ]
```

Also, the function on transport could be renamed to `Transport.metadata(conn, keys)`, `Transport.get_metadata(conn, keys)`, or `Transport.extract_metadata(conn, keys)`.

On top of that, what do you think about using `:metadata` instead of `:meta`?

Reply:
> What do you think about using :meta instead of :extract here?

We chose the word `extract` to try to better illustrate that this data it being extracted from the transport into the socket.

> Also, the function on transport could be renamed to Transport.metadata(conn, keys), Transport.get_metadata(conn, keys), or Transport.extract_metadata(conn, keys).

I like `Transport.extract_metadata(conn, keys)`. I was feeling a little reluctant when I named it just extract :)

> On top of that, what do you think about using :metadata instead of :meta?

`:meta` was selected for consistency in that it is used in other places like in Presence.

Reply:
> We chose the word `extract` to try to better illustrate that this data it being extracted from the transport into the socket.

I see. What do you think then about `:extract_meta` for the websocket configuration, `:meta` for the property and `Transport.extract_meta/2`?